src/Microsoft.AspNet.Mvc.Core/ActionResults/ObjectResult.cs

```diff
@@ -124,20 +140,23 @@ public ObjectResult(object value)
             }
             else
             {
-                // Filter and remove accept headers which cannot support any of the user specified content types.
-                var filteredAndSortedAcceptHeaders = sortedAcceptHeaders
-                                                        .Where(acceptHeader =>
-                                                                ContentTypes
-                                                                    .Any(contentType =>
-                                                                           contentType.IsSubsetOf(acceptHeader)))
-                                                        .ToArray();
-
-                if (filteredAndSortedAcceptHeaders.Length > 0)
+                if (!ignoreAcceptHeader)
                 {
-                    selectedFormatter = SelectFormatterUsingSortedAcceptHeaders(
-                                                                        formatterContext,
-                                                                        formatters,
-                                                                        filteredAndSortedAcceptHeaders);
+                    // Filter and remove accept headers which cannot support any of the user specified content types.
+                    var filteredAndSortedAcceptHeaders = sortedAcceptHeaderMediaTypes
+                                                                    .Where(acceptHeader =>
+                                                                            ContentTypes
+                                                                                .Any(contentType =>
+                                                                                       contentType.IsSubsetOf(acceptHeader)))
+                                                            .ToArray();
```

Comment:
why ToArray and then the Length check? SelectFormatterUsingSortedAcceptHeaders already does all the logic to be efficient